mcmcfit.py

```diff
@@ -1,871 +1,393 @@
+            file_obj.write("  Chisq             = {:.3f}\n".format(
+                model.chisq()))
+
+            print("  ln prior          = {:.3f}".format(model.ln_prior()))
+            file_obj.write("  ln prior          = {:.3f}\n".format(
+                model.ln_prior()))
+
+            print("  ln like           = {:.3f}".format(model.ln_like()))
+            file_obj.write("  ln like           = {:.3f}\n".format(
+                model.ln_like()))
+
+            print("  ln prob           = {:.3f}".format(model.ln_prob()))
+            file_obj.write("  ln prob           = {:.3f}\n".format(
+                model.ln_prob()))
+
+        # Plot the data and the final fit.
+        # model.plot_data(save=True)
+        for iecl in range(neclipses):
+            eclipse = model.search_Node('Eclipse', str(iecl))
+            eclipse.plot_data(save=True)
+
+        exit()
+
+    ##############################################################
+    # MCMC Chain sampler, handled by emcee.                      #
+    # The below plugs the above into emcee's relevant functions. #
+    ##############################################################
+
+    # How many parameters do I have to deal with?
+    npars = len(model.par_val_list)
+
+    print("The MCMC has {:d} variables and {:d} walkers".format(
+        npars, nwalkers))
+    print("(It should have at least 2*npars, {:d} walkers)".format(2*npars))
+
+    # p_0 is the initial position vector of the MCMC walker
+    p_0 = model.par_val_list
+
+    # We cant to scatter that, so create an array of our scatter values.
+    # This will allow us to tweak the scatter value for each individual
+    # parameter.
+    p0_scatter_1 = np.array([scatter_1 for _ in p_0])
+
+    # If comp_scat is asked for, each value wants to be scattered differently.
+    # Some want more, some less.
+    if comp_scat:
+        # scatter factors. p0_scatter_1 will be multiplied by these:
+        scat_fract = {
+            'q':      2,
+            'rwd':    1,
+            'dphi':   0.2,
+            'dFlux':  2,
+            'sFlux':  2,
+            'wdFlux': 2,
+            'rsFlux': 2,
+            'rdisc':  2,
+            'ulimb':  1e-6,
+            'scale':  3,
+            'fis':    3,
+            'dexp':   3,
+            'phi0':   20,
+            'az':     2,
+            'exp1':   5,
+            'exp2':   5,
+            'yaw':    10,
+            'tilt':   2,
+        }
+
+        for par_i, name in enumerate(model.par_names):
+            # Get the parameter of this parName, striping off the node encoding
+            key = name.split("_")[0]
+
+            # Multiply it by the relevant factor
+            p0_scatter_1[par_i] *= scat_fract[key]
+
+        # Create another array for second burn-in
+        p0_scatter_2 = p0_scatter_1*(scatter_2/scatter_1)
+
+    # I need to wrap the model's ln_like, ln_prior, and ln_prob functions
+    # in order to pickle them :(
```

Comment:
agree with the unhappy face. This is annoying

Reply:
>:(